### PR TITLE
Integrate vitcli and update for challenges support

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Implements https://github.com/input-output-hk/jorvit/issues/7

### CODE REVIEW

File: cmd/vitconfig/vitconfig.go [hunk 1]
```diff
@@ -1027,7 +1035,7 @@ func main() {
 		kit.FatalOn(err, "vcli.DbInit", kit.B2S(out))
 
 		// populate the database with already dumped data
-		out, err = vcli.CsvDataLoad(vitDb, fundsFile.Name(), proposalsFile.Name(), votePlansFile.Name())
+		out, err = vcli.CsvDataLoad(vitDb, fundsFile.Name(), proposalsFile.Name(), *challengesPath, votePlansFile.Name())
```
Comment (danielSanchezQ) [inline cmd/vitconfig/vitconfig.go:1031]: As far as I see, there is no need to change anything from the csv itself, so we just need to pipe the path to the vit-servicing-station cli (fund_id is added to the challenges there).
Could you confirm @eugene-babichenko ?

Comment (eugene-babichenko) [reply cmd/vitconfig/vitconfig.go:1031]: Yes, you're right.

Comment (eugene-babichenko) [reply cmd/vitconfig/vitconfig.go:1031]: Shouldn't `err` be processed here or propagated to the caller?

Comment (danielSanchezQ) [reply cmd/vitconfig/vitconfig.go:1031]: It is processed in the next line.

Comment (eugene-babichenko) [reply cmd/vitconfig/vitconfig.go:1031]: Ah, this `kit.FatalOn` tricked me, I expected `if err == nil`.

File: cmd/vitconfig/vitconfig.go [hunk 2]
```diff
@@ -18,15 +18,16 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/input-output-hk/jorvit/pkg/vcli"
+	"github.com/input-output-hk/jorvit/pkg/vstation"
```
Comment (mzabaluev) [inline cmd/vitconfig/vitconfig.go:22]: Could use the same path as the `internal` modules below; I don't see how these modules are special now that they are in tree.

File: cmd/vitconfig/vitconfig.go [hunk 3]
```diff
@@ -50,9 +51,10 @@ type bftLeader struct {
 }
 
 type jcliProposal struct {
-	ExternalID string `json:"external_id"`
-	Options    uint8  `json:"options"`
-	Action     string `json:"action"`
+	ExternalID  string `json:"external_id"`
+	Options     uint8  `json:"options"`
+	Action      string `json:"action"`
+	Challengeid uint32 `json:"challenge_id"`
```
Comment (mzabaluev) [inline cmd/vitconfig/vitconfig.go]: nitpick: `ChallengeID` for consistent capitalization. Or should we be rusty and capitalize as `Id`?

File: cmd/vitconfig/vitconfig.go [hunk 4]
```diff
@@ -1027,7 +1028,7 @@ func main() {
 		kit.FatalOn(err, "vcli.DbInit", kit.B2S(out))
 
 		// populate the database with already dumped data
-		out, err = vcli.CsvDataLoad(vitDb, fundsFile.Name(), proposalsFile.Name(), votePlansFile.Name())
+		out, err = vcli.CsvDataLoad(vitDb, fundsFile.Name(), proposalsFile.Name(), *challengesPath, votePlansFile.Name())
```
Comment (mzabaluev) [inline cmd/vitconfig/vitconfig.go:1031]: One of these is not like the others. I guess `.Name()` does some useful normalization on paths, so it's better used for the challenges file as well?

File: go.mod [hunk 5]
```diff
@@ -5,6 +5,5 @@ go 1.15
 require (
 	github.com/gocarina/gocsv v0.0.0-20201103164230-b291445e0dd2
 	github.com/rinor/jorcli v0.0.0-20201117192102-2a69360d3a83
```
Comment (mzabaluev) [inline go.mod:7]: I guess we'll get to this one in time, too.

File: cmd/vitconfig/vitconfig.go (hunk 2 above)
Comment (danielSanchezQ) [reply cmd/vitconfig/vitconfig.go:22]: Well, vitcli was a "public" API wrapper around the servicing station cli. In Go, whatever public api should go in the pkg folder, hence the decision of keeping it there. If we are gonna maintain it and it is needed externally is better we follow the go styles for this kind of stuff.

File: cmd/vitconfig/vitconfig.go (hunk 3 above)
Comment (danielSanchezQ) [reply cmd/vitconfig/vitconfig.go]: Yep, should be `ChallengeID` for consistency I think 😃

File: cmd/vitconfig/vitconfig.go (hunk 4 above)
Comment (danielSanchezQ) [reply cmd/vitconfig/vitconfig.go:1031]: It is different, the others are actually modified by the script itself, and dumped into another file (those file objects are the ones used to get the new path there with the `.Name()` method).  But in the case of challenges, we just bridge it to the cli itself. It feels a bit overwork opening a file just to use the same syntax.

File: go.mod (hunk 5 above)
Comment (danielSanchezQ) [reply go.mod:7]: Aaaagh, why it doesn't update itself?? 😞

Comment (danielSanchezQ) [reply go.mod:7]: Yeah, we can integrate it in the same way at some point when needed.

**Review by mzabaluev — APPROVED** [review]

**Pull request merged**